Sync core analysis results files to S3

cbethell opened the pull request:
**Issue Addressed**
Closes #277 

**What is the purpose of these changes?**
<!--Provide some background on the changes proposed.-->
This PR modifies the `utils/sync-results.sh` script to also sync the core analysis example results files from our shared directory to a S3 bucket named `scpca-downstream-analyses`.

**What changes did you make?**
<!--Describe the concrete changes that you made or additional features that were added, be as detailed as possible in the steps you took.-->
The `aws sync` command was added at the end of the script and the S3 bucket defined at the top of the script. I was able to successfully run this script to populate the S3 bucket that was created on AWS.

The main `README` is also updated in this PR to include links to the example results files in the S3 bucket.

**Any comments, concerns, or questions important for reviewers**
- Does the public S3 bucket seem to be properly created with the right permissions? For ease of reference: https://s3.console.aws.amazon.com/s3/buckets/scpca-downstream-analyses?region=us-east-1&tab=objects
- Do the links to the files work for you?
- Does this script seem to address all of issue #277?

**Checklist**

Place an `x` in all boxes that you have completed.

- [x] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

allyhawkins commented on utils/sync-results.sh:
```diff
@@ -33,3 +34,5 @@ do
   fi
 done
+
+aws s3 sync $modules_base $s3_base
```
```suggestion
# zip and sync example results
zip -r core_example_results.zip $repo_base/example_results
aws s3 cp example_results.zip $s3_base/example_results.jpg --acl public-read
```
I believe you want to do something like this instead where we zip up the results directory into a single zip file and then copy that file to S3 so that users can use a single link to download an example of the results directory. You also want to make sure that file is publicly accessible.

allyhawkins replied:
You probably want to add a step that removes the local copy of the zip file too after copying.

allyhawkins commented on utils/sync-results.sh:
```diff
@@ -11,6 +11,7 @@ cd ..
 # location for the shared data
 share_base=/shared/data
 modules_base=${share_base}/scpca-downstream-analyses
+s3_base=s3://scpca-downstream-analyses
```
```suggestion
s3_base=s3://scpca-references/example-data/scpca-downstream-analyses
```
Let's actually put these in our public bucket for scpca where we already have stored example results files. I think this makes more sense instead of starting a new bucket.

allyhawkins reviewed (commented):
My first few comments are on making sure that we have the files present on S3 in a way that we can easily link to them. I think we want to have one link that downloads a zip file with the example results. That way then can see the file structure and open individual files if they would like. I added some suggestions to the script to help do that automatically that we should update and then the text can be updated.

cbethell commented:
Thanks for the feedback @allyhawkins! I've renamed the repo variable, zipped and copied over the folder as suggested, and stored it within the scpca-references bucket rather than in a separate bucket (I also deleted the new bucket that I had created)! This is ready for another look!

allyhawkins commented on utils/sync-results.sh:
```diff
@@ -33,3 +34,8 @@ do
+
+# zip and sync example results
+zip -r core_example_results.zip $repo_base/example-results
+aws s3 cp core_example_results.zip $s3_base/example_results.jpg --acl public-read
```
```suggestion
aws s3 cp core_example_results.zip $s3_base/core_example_results.zip --acl public-read
```
This should be `zip` not `jpg`, also let's name it with the full folder name and the core appended to it. I think my thought process here is that we would keep separate zip files with the output from just the core workflow and then each module, otherwise users might get confused seeing all the other output together.

cbethell replied:
Ah yes, that makes total sense 👍

allyhawkins commented on README.md:
```diff
@@ -88,6 +88,7 @@ There are two expected output files thay will be associated with each provided `
     - Clustering that was performed within the workflow
 
 See the [expected output section](#expected-output) for more information on these output files.
+You can also download a ZIP file with examples of the summary HTML report and processed `SingleCellExperiment` objects stored as RDS files [here](https://scpca-references.s3.amazonaws.com/example-data/scpca-downstream-analyses/example_results.jpg).
```
```suggestion
You can also download a ZIP file with an example of the output from running the core workflow, including the summary HTML report and processed `SingleCellExperiment` objects stored as RDS files [here](https://scpca-references.s3.amazonaws.com/example-data/scpca-downstream-analyses/core_example_results.zip).
```
Just a minor rewording suggestion, and renaming the file accordingly based on my comment in the actual script.

allyhawkins commented on README.md:
```diff
@@ -309,6 +310,8 @@ The `<library_id>_core_analysis_report.html` file is the [html file](https://boo
 
 The `<library_id>_processed_sce.rds` file is the [RDS file](https://rstudio-education.github.io/hopr/dataio.html#saving-r-files) that contains the final processed `SingleCellExperiment` object (which contains the filtered, normalized data and clustering results).
 
+You can also download a ZIP file with examples of the summary HTML report and processed `SingleCellExperiment` objects stored as RDS files [here](https://scpca-references.s3.amazonaws.com/example-data/scpca-downstream-analyses/example_results.jpg).
```
```suggestion
You can also download a ZIP file with an example of the output from running the core workflow, including the summary HTML report and processed `SingleCellExperiment` objects stored as RDS files [here](https://scpca-references.s3.amazonaws.com/example-data/scpca-downstream-analyses/core_example_results.zip).
```

allyhawkins reviewed (commented):
These changes look good, just make sure that the file is stored as a zip file on S3. Also we should have a zip file for the example files from each workflow. So the clustering workflow would have a separate zip download and the files should be named with something to distinguish them in s3.

cbethell commented:
_Now_ I think this is just about ready after incorporating those last set of changes!

allyhawkins reviewed (approved):
Looks good! And I checked the file in the bucket and it's set to public so look like everything worked as it should 👍
